File: 2025/day10/clojure/compare.clj
#!/usr/bin/env clojure

(load-file "solution.clj")

(def input (slurp "../input.txt"))
(def lines (clojure.string/split-lines input))

(println "Computing all machines:")
(def results
  (mapv (fn [[i line]]
          (let [[n-counters joltage buttons] (day10.solution/parse-line-part2 line)
                result (day10.solution/solve-machine-part2 n-counters joltage buttons)]
            result))
        (map-indexed vector lines)))

(println "Results:")
(doseq [[i r] (map-indexed vector results)]
  (println (str "Machine " (inc i) ": " r)))

(println "\nTotal:" (reduce + results))

File: 2025/day10/python/compare.py
#!/usr/bin/env python3

from solution import parse_line_part2, solve_machine_part2
from pathlib import Path

input_file = Path(__file__).parent.parent / 'input.txt'
lines = input_file.read_text().strip().split('\n')

print("Computing all machines:")
results = []
for i, line in enumerate(lines, 1):
    n_counters, joltage, buttons = parse_line_part2(line)
    result = solve_machine_part2(n_counters, joltage, buttons)
    results.append(result)
    if i == 183:
        print(f"Machine {i}: {result} (joltage={joltage}, buttons={buttons})")

print(f"\nTotal: {sum(results)}")
